Expand websocket closure code handling

cosive-james-cooper opened the pull request:
This commit introduces a substantially expanded error handling block for when the websocket has closed unexpectedly, leading to a timeout error.  Specifically, now all 14 defined closure status codes are handled individually, rather than four of them.  Furthermore, the catch-all exception handler at the end has been modified to provide (slightly) more information that should hopefully be more useful to us in debugging errors on production boxes.

This also includes a bunch of small tidying measures taken in the chrome.py file, and adjusted the initial declaration location of a couple of variables, since it was possible those variables could be used without having first been declared.

Lastly, the requirements.txt files have been updated.  In requirements.txt the lxml requirement has been updated to a more recent version, while dev_requirements.txt has been changed to include requirements.txt in itself explicitly.

There will probably be several 'changes' in this PR that relate to whitespace only, so you might want to turn on the "no whitespace" option here.

cosive-james-cooper committed:
Expand websocket closure code handling

This commit introduces a substantially expanded error handling block for when the websocket has closed unexpectedly, leading to a timeout error.  Specifically, now all 14 defined closure status codes are handled individually, rather than four of them.  Furthermore, the catch-all exception handler at the end has been modified to provide (slightly) more information that should hopefully be more useful to us in debugging errors on production PhishFeeder boxes.

This also includes a bunch of small tidying measures taken in the chrome.py file, and adjusted the initial declaration location of a couple of variables, since it was possible those variables could be used without having first been declared.

Lastly, the requirements.txt files have been updated.  In requirements.txt the lxml requirement has been updated to a more recent version, while dev_requirements.txt has been changed to include requirements.txt in itself explicitly.

cosive-james-cooper committed:
Expand websocket closure code handling

This commit introduces a substantially expanded error handling block for when the websocket has closed unexpectedly, leading to a timeout error.  Specifically, now all 14 defined closure status codes are handled individually, rather than four of them.  Furthermore, the catch-all exception handler at the end has been modified to provide (slightly) more information that should hopefully be more useful to us in debugging errors on production PhishFeeder boxes.

This also includes a bunch of small tidying measures taken in the chrome.py file, and adjusted the initial declaration location of a couple of variables, since it was possible those variables could be used without having first been declared.

Lastly, the requirements.txt files have been updated.  In requirements.txt the lxml requirement has been updated to a more recent version, while dev_requirements.txt has been changed to include requirements.txt in itself explicitly.

cosive-james-cooper commented:
Forgot to mention...  This should close Jira issue AUTO-515 if/when merged.

cosive-james-cooper committed:
Update setup.py

Update setup.py to reflect the development of this branch of Chromewhip by Cosive, and better reflect the current status of our fork.  Also, bring the stated requirements listed in setup.py into line with those found in requirements.txt.  Plus, fix what appeared to be a few small errors in the development scripts.

cosive-james-cooper committed:
Update setup.py

Update setup.py to reflect the development of this branch of Chromewhip by Cosive, and better reflect the current status of our fork.  Also, bring the stated requirements listed in setup.py into line with those found in requirements.txt.  Plus, fix what appeared to be a few small errors in the development scripts.

cosive-james-cooper committed:
Update Chrome Devtools protocol to match Chrome 96.0.4664.110

Following the semi-automated process established to run these things, this commit updates the Chromewhip protocol to match that produced by a call to chrome's json/protocol on the latest version of the chromewhip docker container.

Unfortunately, I didn't manage to work out which version of the protocol this is.

This commit also adjusts some of the numbering in browser_protocol_patch.json.

cosive-james-cooper committed:
Merge pull request #8 from Cosive/chrome-96.0.4664.110

cosive-james-cooper committed:
Merge remote-tracking branch 'origin/AUTO-515-insufficient-detail-in-chromewhip-reporting-of-web-socket-closures' into AUTO-515-insufficient-detail-in-chromewhip-reporting-of-web-socket-closures

cosive-james-cooper commented:
For what it is worth, we now seem to have been through several rounds of successful testing using the head commit on this branch.  I propose we consider again merging this back into master :)

cosive-james-cooper committed:
Bump version in setup.py

This commit just bumps the stated version of Chromewhip in setup.py from 0.3.4 to 0.3.5 (something I forgot to do earlier).  It also updates the download_url field to point to the latest rolled version we have, which is currently 96.0.4664.110.

ahosie commented on setup.py:
```diff
@@ -7,27 +7,32 @@
-    url='https://github.com/chuckus/chromewhip',
-    download_url='https://github.com/chuckus/chromewhip/archive/v0.3.4.tar.gz',
+    url='https://github.com/Cosive/chromewhip',
+    download_url='https://github.com/Cosive/chromewhip/archive/refs/tags/chrome-96.0.4664.110.tar.gz',
```
This download_url will be brittle, and we're not publishing this module to PyPI, so there's no value added here.

I strongly recommend leaving this (and all files sourced from a third-party repo) as-is - unmodified from the original source - unless a change is strictly required to minimise merge fights. Alternatively a (more) stable download URL could be set.

```suggestion
    download_url='https://codeload.github.com/Cosive/chromewhip/tar.gz/refs/heads/master',
```

cosive-james-cooper replied:
I'm moderately certain that the old download URL listed here was the one that was actually being used when downloading Chromewhip for use in other software (I can't remember precisely why I think that now, though).  Thus, it was quite important to update the URL from what it had been, i.e. an old one on the repo we don't control.

Your proposed new URL does seem likely to be the better choice, though.

ahosie commented on setup.py:
```diff
@@ -7,27 +7,32 @@
 
     description='asyncio driver + HTTP server for Chrome devtools protocol',
     long_description=readme(),
+    # This project was originally forked from https://github.com/chuckus/chromewhip, but the version included here
+    # has been developed subsequently by Cosive, while the original version doesn't appear to have been maintained in
+    # some time.
```
```suggestion

```

cosive-james-cooper committed:
Update setup.py

Change the download URL to the proposed one that points to the head of the master branch.

Co-authored-by: ahosie <ahosie+github@gmail.com>

cosive-james-cooper committed:
Update setup.py

Co-authored-by: ahosie <ahosie+github@gmail.com>
